test/integration/migrate/index.js

```diff
@@ -268,12 +221,52 @@ module.exports = function(knex) {
 
   });
 
+  describe('knex.migrate.latest in parallel', function() {
+
+    afterEach(function() {
+      return knex.migrate.rollback({directory: 'test/integration/migrate/test'});
+    });
+
+    if (knex.client.dialect === 'postgres') {
+      it("is able to run two migrations in parallel in postgres", function () {
```

Comment:
The default transaction modes in the other databases either do auto-commit (e.g. MySQL) or don't support `for share` locking (e.g. SQLite). Therefore (by default) concurrent migrations are expected to fail in the same way as before for the non-Postgres databases.